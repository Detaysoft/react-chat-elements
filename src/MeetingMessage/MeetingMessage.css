.rce-mbox-mtmg {
    display: flex;
    justify-content: center;
    align-content: center;
    padding-bottom: 13px;
    min-width: 425px;
    max-width: 425px;
}

.rce-mtmg {
    width: 100%;
    position: relative;
    background: #f5f5f5;
    display: flex;
    flex-direction: column;
    margin: 5px 0px;
    float: left;
    border-radius: 2px;
}

.rce-mtmg-title {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    width: 300px;
}

.rce-mtmg-subject {
    text-align: start;
    display: inline-block;
    font-size: 15px;
    padding: 5px 9px;
}

.rce-mtmg-toogleItem {
    width: 100%;
    height: 100%;
}

.rce-mtmg-body {
    height: 50px;
    background: #6264a7;
    color: white;
    font-size: 15px;
    display: flex;
    justify-content: space-between;
}

.rce-mtmg-body:hover {
    opacity: 0.9;
}

.rce-mtmg-item {
    display: flex;
    align-items: center;
    justify-content: flex-start;
    padding: 0px 10px;
}

.rce-mtmg-item > svg {
    width: 23px;
    height: 23px;
}

.rce-mtmg-content {
    display: flex;
    flex-direction: column;
    padding: 0 10px;
}

.rce-mtmg-date {
    color: #cecece;
    font-size: 13px;
}

.rce-mtmg-body-bottom {
    display: flex;
    flex-direction: row;
    padding: 9px;
    color: #6264a7;
    cursor: pointer;
    font-size: 13px;
}

.rce-mtmg-bottom--tptitle {
    display: flex;
    justify-content: center;
    align-items: center;
}

.rce-mtmg-bottom--tptitle > svg,
.rce-mtmg-body-bottom--bttitle > svg {
    color: #6264a7a1;
}

.rce-mtmg-toogleContent {
    display: none;
    height: auto;
    min-height: 60px;
    flex-direction: column;
}

.rce-mtmg-toogleContent--click {
    display: flex;
}

.rce-mtmg-right-icon {
    display: flex;
    align-items: center;
    position: relative;
    right: 10px;
    cursor: pointer;
}

.rce-mtmg-right-icon > svg {
    width: 23px;
    height: 23px;
}

.rce-mitem {
    display: flex;
    padding: 10px 8px;
}

.rce-mitem:hover,
.rce-mitem-event:hover {
    background: #ececec;
}

.rce-mitem-event {
    user-select:  none;
}

.rce-mitem-body {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    flex-direction: column;
}

.rce-mitem-body-content {
    display: flex;
    align-items: flex-start;
    justify-content: flex-start;
}

.rce-mitem-body--top {
    display: flex;
    align-items: center;
    margin-bottom: 2px;
    user-select: none;
}

.rce-mitem-body--top-title {
    font-size: 15px;
    color: #6264a7;
    padding: 0px 15px 0 0;
    text-transform: capitalize;
    font-weight: 600;
}

.rce-mitem-body--top-title:hover {
    cursor: pointer;
    text-decoration: underline;
}

.rce-mitem-body--bottom-title {
    color: #252525;
    font-size: 13px;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

.rce-mitem-body--top-time {
    font-size: 12px;
    color: rgba(0,0,0,0.4);
}

.rce-mitem-bottom-body {
    padding: 10px 8px;
    display: flex;
    align-items: flex-start;
    justify-content: flex-start;
}

.rce-mitem-body.avatar {
    padding: 8px 10px 0 0;
}

.rce-mitem.avatar {
    padding: 8px 5px 0 0;
}

.rce-mitem.no-avatar {
    padding: 8px 10px 0 0;
}

.rce-mitem.no-avatar > svg {
    width: 19px;
    height: 19px;
}

.rce-mitem.avatar img {
    width: 22px;
    height: 22px;
    border: none !important;
    background: #ccc;
    border-radius: 100%;
}

.rce-mitem-body.avatar > svg {
    width: 19px;
    height: 19px;
}

.rce-mitem-bottom-body-top {
    display: flex;
    flex-direction: column;
}

.rce-mitem-bottom-body-top-title > svg {
    padding: 0 7px 0 0;
}

.rce-mitem-avatar-content {
    position: absolute;
    right: 10px;
    display: flex;
}

.rce-mitem-avatar {
    padding: 0 3px 0 0;
}

.rce-mitem-avatar img {
    width: 50px;
    height: 50px;
    border: none !important;
    background: #ccc;
    border-radius: 100%;
}

.rce-mtmg-call-record {
    width: 350px;
    height: 85%;
    background: #eaeaea;
    margin-top: 11px;
}

.rce-mtmg-call-body {
    display: flex;
    align-items: flex-start;
    flex-direction: row;
    position: relative;
}

.rce-mtmg-call-avatars {
    width: 140px !important;
    height: 100px !important;
    position: relative;
}

.rce-mtmg-call-avatars img {
    width: 100% !important;
    height: 100% !important;
    background: #ccc;
    cursor: pointer;
}

.rce-mtmg-call-body-title {
    display: flex;
    position: relative;
    flex-direction: column;
    top: 30px;
    left: 15px;
}

.rce-mtmg-call-body-title > span {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    width: 185px;
}

.rce-mtmg-call-body-bottom {
    color: #505050;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    width: 185px;
    font-size: 12px;
}

.rce-mtmg-record-time {
    position: absolute;
    bottom: 5px;
    right: 5px;
    font-size: 13px;
    background: #000000cf;
    color: white;
    padding: 4px;
    border-radius: 5px;
}
