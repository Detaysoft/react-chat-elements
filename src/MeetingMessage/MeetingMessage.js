import React, { Component } from 'react';
import './MeetingMessage.css';

import FaCalendar from 'react-icons/lib/fa/calendar';
import MdMoreHoriz from 'react-icons/lib/md/more-horiz';
import FaCaretDown from 'react-icons/lib/fa/caret-down';
import FaCaretRight from 'react-icons/lib/fa/caret-right';
import IoVideo from 'react-icons/lib/io/ios-videocam-outline';
import IoChatboxes from 'react-icons/lib/io/ios-chatboxes-outline';

import {
    format,
} from'timeago.js';

import Avatar from '../Avatar/Avatar';

import classNames from 'classnames';

export class MeetingMessage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            toogle: false,
        };
    }

    toggleClick() {
        this.setState({
            toogle: !this.state.toogle,
        })
    }

    render() {
        const {
            date,
            dateString,
            title,
            subject,
            onClick,
            collapseTitle,
            dataSource,
            participants,
            onMeetingTitleClick,
            onMeetingVideoLinkClick,
        } = this.props;

        const PARTICIPANT_LIMIT = this.props.participantsLimit;

        const dateText = dateString ? dateString : (date && !isNaN(date) && (format(date)));

        return (
            <div className="rce-mbox-mtmg">
                <div className="rce-mtmg">
                    <div
                        className="rce-mtmg-subject">
                        {subject || 'Unknown Meeting'}
                    </div>
                    <div
                        className="rce-mtmg-body"
                        onClick={onClick}>
                        <div className="rce-mtmg-item">
                            <FaCalendar />
                            <div className="rce-mtmg-content">
                                <span className="rce-mtmg-title">
                                    {title}
                                </span>
                                <span className='rce-mtmg-date'>
                                    {dateText}
                                </span>
                            </div>
                        </div>
                        <div className="rce-mtmg-right-icon">
                            <MdMoreHoriz/>
                        </div>
                    </div>
                    <div
                        className="rce-mtmg-body-bottom"
                        onClick={() => this.toggleClick()}>
                        {
                            this.state.toogle === true ?
                            <div className="rce-mtmg-bottom--tptitle">
                                <FaCaretDown/>
                                <span>{collapseTitle}</span>
                            </div>
                            :
                            <div className="rce-mtmg-body-bottom--bttitle">
                                <FaCaretRight/>
                                <span>
                                    {
                                        participants.slice(0, PARTICIPANT_LIMIT).map(x => x.title || 'Unknow').join(', ')
                                    }
                                    {
                                        participants.length > PARTICIPANT_LIMIT &&
                                            `, +${(participants.length - PARTICIPANT_LIMIT)}`
                                    }
                                </span>
                            </div>
                        }
                    </div>
                    <div
                        className={classNames(
                            'rce-mtmg-toogleContent',
                            {'rce-mtmg-toogleContent--click': this.state.toogle === true}
                        )}>
                        {
                            dataSource &&
                            dataSource.map((x, i) => {
                                return (
                                    <div key={i}>
                                        {
                                            !x.event &&
                                            <div className="rce-mitem">
                                                <div
                                                    className={classNames(
                                                        'rce-mitem avatar',
                                                        {'rce-mitem no-avatar': !x.avatar}
                                                    )}>
                                                    {
                                                        x.avatar ?
                                                            <Avatar src={x.avatar}/>
                                                            : <IoChatboxes />
                                                    }
                                                </div>
                                                <div className="rce-mitem-body">
                                                    <div className="rce-mitem-body--top">
                                                        <div
                                                            className="rce-mitem-body--top-title"
                                                            onClick={(e) => onMeetingTitleClick(x, i, e)}>
                                                            {x.title}
                                                        </div>
                                                        <div className="rce-mitem-body--top-time">
                                                            {
                                                                x.dateString ? x.dateString : (x.date &&
                                                                !isNaN(x.date) &&
                                                                (format(x.date)))
                                                            }
                                                        </div>
                                                    </div>
                                                    <div className="rce-mitem-body--bottom">
                                                        <div
                                                            className="rce-mitem-body--bottom-title" >
                                                            {x.message}
                                                        </div>
                                                    </div>
                                                </div>
                                            </div>
                                        }

                                        {
                                            x.event &&
                                            <div className="rce-mitem-event">
                                                <div className="rce-mitem-bottom-body">
                                                    <div className="rce-mitem-body avatar">
                                                        <IoVideo />
                                                    </div>
                                                    <div className="rce-mitem-bottom-body-top">
                                                        {x.event.title}
                                                        <div className="rce-mitem-body--top-time">
                                                            {
                                                                x.dateString ? x.dateString : (x.date &&
                                                                !isNaN(x.date) &&
                                                                (format(x.date)))
                                                            }
                                                        </div>
                                                        <div className="rce-mitem-avatar-content">
                                                            {
                                                                x.event.avatars &&
                                                                x.event.avatars.map((x, i) => x instanceof Avatar ? x : (
                                                                    <div key={i} className="rce-mitem-avatar">
                                                                        <Avatar
                                                                            src={x.src}/>
                                                                    </div>
                                                                ))
                                                            }
                                                        </div>
                                                        {
                                                            x.record &&
                                                            <div className="rce-mtmg-call-record">
                                                                <div className="rce-mtmg-call-body">
                                                                    <div
                                                                        onClick={(e) => onMeetingVideoLinkClick(x, i, e)}
                                                                        className="rce-mtmg-call-avatars">
                                                                        <Avatar
                                                                            className={'rce-mtmg-call-avatars'}
                                                                            src={x.record.avatar}/>
                                                                        <div className={'rce-mtmg-record-time'}>
                                                                            {x.record.time}
                                                                        </div>
                                                                    </div>
                                                                    <div className="rce-mtmg-call-body-title">
                                                                        <span>
                                                                            {x.record.title}
                                                                        </span>
                                                                        <div className="rce-mtmg-call-body-bottom">
                                                                            {x.record.savedBy}
                                                                        </div>
                                                                    </div>
                                                                </div>
                                                            </div>
                                                        }
                                                    </div>
                                                </div>
                                            </div>
                                        }
                                    </div>
                                )
                            })
                        }
                    </div>
                </div>
            </div>
        );
    }
}

MeetingMessage.defaultProps = {
    date: new Date(),
    dateString: '',
    title: '',
    subject: '',
    collapseTitle: '',
    participantsLimit: 3,
    avatarFlexible: false,
    dataSource: [],
    participants: [],
    onClick: () => void(0),
    onMeetingTitleClick: () => void(0),
    onMeetingVideoLinkClick: () => void(0),
    onAvatarError: () => void(0),
};

export default MeetingMessage;
