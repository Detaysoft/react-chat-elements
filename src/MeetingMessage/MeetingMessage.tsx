import React, { FC, useState } from 'react'
import './MeetingMessage.css'

import { FaCalendar, FaCaretDown, FaCaretRight } from 'react-icons/fa'
import { HiOutlineVideoCamera } from 'react-icons/hi'
import { IoMdChatboxes } from 'react-icons/io'
import { MdMoreHoriz } from 'react-icons/md'

import { format } from 'timeago.js'

import Avatar from '../Avatar/Avatar'
import Dropdown from '../Dropdown/Dropdown'

import classNames from 'classnames'
import { IMeetingMessage, IMeetingMessageProps, MeetingMessageEvent} from '../type'

const MeetingMessage: FC<IMeetingMessageProps> = ({
                                                      date,
                                                      dateString,
                                                      title,
                                                      subject,
                                                      collapseTitle,
                                                      moreItems,
                                                      participants,
                                                      dataSource,

                                                      onClick,
                                                      onMeetingTitleClick,
                                                      onMeetingVideoLinkClick,
                                                      onMeetingMoreSelect,
                                                      ...props
                                                  }) => {
    const [toogle, setToogle] = useState(false)

    const PARTICIPANT_LIMIT = props.participantsLimit
    const dateText = dateString ? dateString : date && format(date)

    const _onMeetingLinkClick: MeetingMessageEvent = (item: IMeetingMessage, index: number, event: React.MouseEvent<HTMLElement, MouseEvent>) => {
    if (onMeetingTitleClick instanceof Function) onMeetingTitleClick(item, index, event)
  }

  const _onMeetingVideoLinkClick: MeetingMessageEvent = (item: IMeetingMessage, index: number, event: React.MouseEvent<HTMLElement, MouseEvent>) => {
    if (onMeetingVideoLinkClick instanceof Function) onMeetingVideoLinkClick(item, index, event)
  }

  const toggleClick = () => {
    setToogle(!toogle)
  }

  return (
    <div className='rce-mbox-mtmg'>
      <div className='rce-mtmg'>
        <div className='rce-mtmg-subject'>{subject || 'Unknown Meeting'}</div>
        <div className='rce-mtmg-body' onClick={onClick}>
          <div className='rce-mtmg-item'>
            <FaCalendar />
            <div className='rce-mtmg-content'>
              <span className='rce-mtmg-title'>{title}</span>
              <span className='rce-mtmg-date'>{dateText}</span>
            </div>
          </div>

          {onMeetingMoreSelect && moreItems && moreItems.length > 0 && (
            <div>
              <Dropdown
                animationType='bottom'
                animationPosition='norteast'
                buttonProps={{
                  className: 'rce-mtmg-right-icon',
                  icon: {
                    component: <MdMoreHoriz />,
                    size: 24,
                  },
                }}
                items={moreItems}
                onSelect={onMeetingMoreSelect}
              />
            </div>
          )}
        </div>
        <div className='rce-mtmg-body-bottom' onClick={toggleClick}>
          {toogle === true ? (
            <div className='rce-mtmg-bottom--tptitle'>
              <FaCaretDown />
              <span>{collapseTitle}</span>
            </div>
          ) : (
            <div className='rce-mtmg-body-bottom--bttitle'>
              <FaCaretRight />
              <span>
                {participants
                  ?.slice(0, PARTICIPANT_LIMIT)
                  .map(x => x.title || 'Unknow')
                  .join(', ')}
                {participants &&
                  PARTICIPANT_LIMIT &&
                  participants.length > PARTICIPANT_LIMIT &&
                  `, +${participants.length - PARTICIPANT_LIMIT}`}
              </span>
            </div>
          )}
        </div>
        <div className={classNames('rce-mtmg-toogleContent', { 'rce-mtmg-toogleContent--click': toogle === true })}>
          {dataSource &&
            dataSource.map((x, i) => {
              return (
                <div key={i}>
                  {!x.event && (
                    <div className='rce-mitem'>
                      <div className={classNames('rce-mitem avatar', { 'rce-mitem no-avatar': !x.avatar })}>
                        {x.avatar ? <Avatar src={x.avatar} /> : <IoMdChatboxes />}
                      </div>
                      <div className='rce-mitem-body'>
                        <div className='rce-mitem-body--top'>
                          <div
                            className='rce-mitem-body--top-title'
                            onClick={(e: React.MouseEvent<HTMLElement>) => _onMeetingLinkClick(x, i, e)}
                          >
                            {x.title}
                          </div>
                          <div className='rce-mitem-body--top-time'>
                            {x.dateString ? x.dateString : x.date && x.date && format(x.date)}
                          </div>
                        </div>
                        <div className='rce-mitem-body--bottom'>
                          <div className='rce-mitem-body--bottom-title'>{x.message}</div>
                        </div>
                      </div>
                    </div>
                  )}
                  {x.event && (
                    <div className='rce-mitem-event'>
                      <div className='rce-mitem-bottom-body'>
                        <div className='rce-mitem-body avatar'>
                          <HiOutlineVideoCamera />
                        </div>
                        <div className='rce-mitem-bottom-body-top'>
                          {x.event.title}
                          <div className='rce-mitem-body--top-time'>{x.dateString ? x.dateString : x.date && format(x.date)}</div>
                          <div className='rce-mitem-avatar-content'>
                            {
                              <div className='rce-mitem-avatar'>
                                {x.event.avatars &&
                                  // x.event.avatars.slice(0, x.event.avatarsLimit).map((x, i) => x instanceof Avatar ? x : (
                                  x.event.avatars.slice(0, x.event.avatarsLimit).map((x, i) => <Avatar key={i} src={x.src} />)}
                                {x.event.avatars && x.event.avatarsLimit && x.event.avatars.length > x.event.avatarsLimit && (
                                  <div
                                    className='rce-mitem-length rce-mitem-tooltip'
                                    tooltip={x.event.avatars
                                      .slice(x.event.avatarsLimit, x.event.avatars.length)
                                      .map(avatar => avatar.title)
                                      .join(',')
                                      .toString()}
                                  >
                                    <span className='rce-mitem-tooltip-text'>
                                      {'+' + (x.event.avatars.length - x.event.avatarsLimit)}
                                    </span>
                                  </div>
                                )}
                              </div>
                            }
                          </div>
                          {x.record && (
                            <div className='rce-mtmg-call-record'>
                              <div className='rce-mtmg-call-body'>
                                <div
                                  onClick={(e: React.MouseEvent<HTMLElement>) => _onMeetingVideoLinkClick(x, i, e)}
                                  className='rce-mtmg-call-avatars'
                                >
                                  <Avatar className={'rce-mtmg-call-avatars'} src={x.record.avatar} />
                                  <div className={'rce-mtmg-record-time'}>{x.record.time}</div>
                                </div>
                                <div className='rce-mtmg-call-body-title'>
                                  <span>{x.record.title}</span>
                                  <div className='rce-mtmg-call-body-bottom'>{x.record.savedBy}</div>
                                </div>
                              </div>
                            </div>
                          )}
                        </div>
                      </div>
                    </div>
                  )}
                </div>
              )
            })}
        </div>
      </div>
    </div>
  )
}

export default MeetingMessage
