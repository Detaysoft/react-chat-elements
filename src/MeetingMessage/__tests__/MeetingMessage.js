import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MeetingMessage from '../MeetingMessage';

describe('MeetingMessage component', () => {
  it('should render without issues', () => {
    const component = shallow(<MeetingMessage />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
