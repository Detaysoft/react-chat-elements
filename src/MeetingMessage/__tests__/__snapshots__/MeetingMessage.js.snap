// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingMessage component should render without issues 1`] = `
<div
  className="rce-mbox-mtmg"
>
  <div
    className="rce-mtmg"
  >
    <div
      className="rce-mtmg-subject"
    >
      Unknown Meeting
    </div>
    <div
      className="rce-mtmg-body"
      onClick={[Function]}
    >
      <div
        className="rce-mtmg-item"
      >
        <FaCalendar />
        <div
          className="rce-mtmg-content"
        >
          <span
            className="rce-mtmg-title"
          />
          <span
            className="rce-mtmg-date"
          >
            just now
          </span>
        </div>
      </div>
      <div
        className="rce-mtmg-right-icon"
      >
        <MdMoreHoriz />
      </div>
    </div>
    <div
      className="rce-mtmg-body-bottom"
      onClick={[Function]}
    >
      <div
        className="rce-mtmg-body-bottom--bttitle"
      >
        <FaCaretRight />
        <span />
      </div>
    </div>
    <div
      className="rce-mtmg-toogleContent"
    />
  </div>
</div>
`;
